### PR TITLE
Change javascript implementation to typescript

### PR DESCRIPTION
**Pull request by Powerade**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -51,3 +51,5 @@ buck-out/
 */fastlane/report.xml
 */fastlane/Preview.html
 */fastlane/screenshots
+
+.editorconfig
```
Comment (ivangabriele) [inline .gitignore]: You don't have to ignore the `.editorconfig`, it's a shared normalization config.

File: tslint.json [hunk 2]
```diff
@@ -0,0 +1,136 @@
+{
```
Comment (ivangabriele) [inline tslint.json:1]: You don't have to configure all that manually, it would be endless to do so ;) tslint has some full default config from which you can inherit and eventually disable some not-so-important rules.

Check how I use the `extends` [there](https://github.com/Electra-project/Electra-JS/blob/master/tslint.json).

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (ivangabriele) [inline tsconfig.json:1]: Try to order you props by alphabetical order, it's easier to read. At least the `compilerOptions` ones.

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        "noImplicitThis": true,
+        "noImplicitAny": true,
+        "strictNullChecks": true,
+        "lib": [
+            "dom",
+            "es2017"
+        ],
+        "noUnusedLocals": true
+    },
+    "filesGlob": [
```
Comment (ivangabriele) [inline tsconfig.json]: No need for this prop.

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{
+    "compilerOptions": {
+        "target": "es2015",
```
Comment (ivangabriele) [inline tsconfig.json]: You can target `es2016`, it should be perfectly compatible.

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        "forceConsistentCasingInFileNames": true,
+        "noImplicitReturns": true,
+        "noUnusedParameters": true,
+        "noEmit": false,
+        "skipLibCheck": true,
+        "noImplicitThis": true,
+        "noImplicitAny": true,
+        "strictNullChecks": true,
+        "lib": [
+            "dom",
```
Comment (ivangabriele) [inline tsconfig.json]: I'm not 100% sure about that, but React Native is not a embedded webview like Cordova, so I think the dom lib is not supported. We shouldn't need it in any case.

File: dist/index.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { TabNavigator } from 'react-navigation';
```
Comment (ivangabriele) [inline dist/index.js]: Don't include the dist build in your dev commits. When you run a build to regenerate the dist files, do it as a separate commit called something like: `build: update`.

**Review by ivangabriele — COMMENTED** [review]
Some global comments:

- No need for semicolons, it creates more code noise. Semicolons were necessary more than a decade ago because of some JS engines inconsistencies and bad coding practices (like one-lined multiple expressions and so on).
- It's better to order style props in alphabetical order: when the code will grow, it's faster to "catch" the props we are looking for.

File: tsconfig.json (hunk 3 above)
Comment (Powerade) [reply tsconfig.json:1]: done

File: tslint.json (hunk 2 above)
Comment (Powerade) [reply tslint.json:1]: done

**Review by ivangabriele — APPROVED** [review]
Good job !
